xbmc/music/MusicDatabase.cpp

```diff
@@ -3600,8 +3600,8 @@ bool CMusicDatabase::GetRolesNav(const std::string& strBaseDir, CFileItemList& i
       return false;
 
     // get roles with artists having that role
-    std::string strSQL = "SELECT idRole, strRole FROM role WHERE EXISTS "
-                         "(SELECT 1 FROM song_artist WHERE song_artist.idRole = role.idRole)";
+    std::string strSQL = "SELECT DISTINCT role.idRole, role.strRole FROM role "
+                         "JOIN song_artist ON song_artist.idRole = role.idRole ";
```

Comment:
Yes they are. When I tested them on SQLite the join was unexpectedly quicker than the exists, I guess it hits some index. But to be honest my testing was not extensive, and a different data combination could change things too, and MySQL could be different again. So I would go with the join, but it is pretty arbitary. But many thanks for the interest.